src/test/java/marquez/api/models/JobRequestTest.java

```diff
@@ -56,4 +66,38 @@ public void testNewRequest_noDescription() {
             INPUT_DATASET_URNS, OUTPUT_DATASET_URNS, LOCATION_VALUE, NO_DESCRIPTION_VALUE);
     assertThat(actual).isEqualTo(expected);
   }
+
+  @Test
+  public void testNewRequest_fromJson() throws Exception {
+    final ArrayNode array0 = MAPPER.valueToTree(REQUEST.getInputDatasetUrns());
+    final ArrayNode array1 = MAPPER.valueToTree(REQUEST.getOutputDatasetUrns());
+    final ObjectNode obj =
+        MAPPER
+            .createObjectNode()
+            .put("location", REQUEST.getLocation())
+            .put("description", REQUEST.getDescription().get());
+    obj.putArray("inputDatasetUrns").addAll(array0);
+    obj.putArray("outputDatasetUrns").addAll(array1);
+
+    final String requestAsJson = obj.toString();
+    final JobRequest actual = MAPPER.readValue(requestAsJson, JobRequest.class);
+    assertThat(actual).isEqualTo(REQUEST);
+  }
+
+  @Test
+  public void testNewRequest_fromJson_noDescription() throws Exception {
+    final ArrayNode array0 = MAPPER.valueToTree(REQUEST_NO_DESCRIPTION.getInputDatasetUrns());
+    final ArrayNode array1 = MAPPER.valueToTree(REQUEST_NO_DESCRIPTION.getOutputDatasetUrns());
+    final ObjectNode obj =
+        MAPPER
+            .createObjectNode()
+            .put("location", REQUEST_NO_DESCRIPTION.getLocation())
+            .put("description", REQUEST_NO_DESCRIPTION.getDescription().orElse(null));
```

Comment:
I'm wondering what the behavior will be in the case that description is `null`. From what I understand, the simulated json here will contain the description field, with the value `null`. This is great for testing machine-to-machine interactions, where if there is no description, it'll just be set to `null`.

However, I'm concerned that this isn't testing the case where a user just omits the field altogether, because by default, Jackson still serialized fields that are null.

Reply:
It's a fair point. But as I mentioned in the response above (see [link](https://github.com/MarquezProject/marquez/pull/600#discussion_r308016493)), we'd be testing how Jackson handles such cases. Also, `Optional` fields already default to _null_.